elements/pfelement/src/pfelement.js

```diff
@@ -19,14 +19,21 @@ class PFElement extends HTMLElement {
   /**
    * A boolean value that indicates if the logging should be printed to the console; used for debugging.
    * For use in a JS file or script tag; can also be added in the constructor of a component during development.
-   * @example PFElement._debugLog = true;
+   * @example PFElement.debugLog(true);
    * @tags debug
    */
   static debugLog(preference = null) {
     if (preference !== null) {
-      PFElement._debugLog = !!preference;
+      // wrap localStorage references in a try/catch; merely referencing it can
+      // throw errors in some locked down environments
+      try {
+        localStorage.pfeLog = !!preference;
+      } catch (e) {
+        return false;
```

Comment:
If local storage fails, do we want to fallback to PFElement._debugLog?
```suggestion
        return PFElement._debugLog || false;
```

Reply:
Sure, that sounds good.  Related, I considered having the error case return `true` so that in environments with busted localStorage, the logs are always on.  Then I decided that's a far-flung of an edge case, that we're not likely to ever run into (I've only seen it once, years ago).  (Famous last words...)

Reply:
All done!